common/views/pages/_document.tsx

```diff
@@ -16,14 +16,63 @@ import {
   GoogleTagManagerNoScript,
   GaDimensions,
 } from '../../services/app/google-analytics';
-import { getCookies } from 'cookies-next';
-import { getConsentCookieServerSide } from '@weco/common/utils/cookie-consent';
+import { getConsentState } from '@weco/common/utils/cookie-consent';
 
 const {
   ANALYTICS_WRITE_KEY = '78Czn5jNSaMSVrBq2J9K4yJjWxh6fyRI',
   NODE_ENV = 'development',
 } = process.env;
 
+const CookieControl: FunctionComponent = () => {
+  // TODO: I'd expect this to shout when the event is fired, but it doesn't :\
+  useEffect(() => {
+    const sayHello = () => {
+      console.log('HEELLO');
+    };
+
+    window.addEventListener('analyticsConsentWithdrawn', sayHello);
+
+    return () => {
+      window.removeEventListener('analyticsConsentWithdrawn', sayHello);
+    };
+  });
+
+  return (
+    <>
+      <script
+        src="https://cc.cdn.civiccomputing.com/9/cookieControl-9.x.min.js"
+        type="text/javascript"
+      ></script>
+      <script
+        dangerouslySetInnerHTML={{
+          __html: `CookieControl.load({
+                product: 'COMMUNITY',
+                apiKey: '73fee8f69cf633d66fae404ddd69d2559af7f887',
+                necessaryCookies: ['toggle_*'],
+                optionalCookies: [
+                  {
+                      name: 'analytics',
+                      label: 'Google Analytics',
+                      description: 'Analytical cookies help us to improve our website by collecting and reporting information on its usage.',
+                      cookies: ['_ga', '_ga*', '_gid', '_gat', '__utma', '__utmt', '__utmb', '__utmc', '__utmz', '__utmv'],
+                      onAccept: function(){
+                          console.log('analytics accepted');
+                      },
+                      onRevoke: function(){
+                          console.log('analytics rejected');
+                          // TODO: Hoping we can listen for this in a component and react to it!
+                          const event = new CustomEvent('analyticsConsentWithdrawn', {});
+                          window.dispatchEvent(event);
```

Comment:
I can add a listener via the console and see this event being fired, so this bit works.